src/test/java/de/adesso/kicker/user/UserControllerTest.java

```diff
@@ -75,4 +78,33 @@ void whenUserExistsReturnUser() throws Exception {
                 .andExpect(model().attribute("rankingPosition", 1));
     }
 
+    @Test
+    void whenUserNotExistentExpectNotFound() throws Exception {
+        // given
+        willThrow(UserNotFoundException.class).given(userService).getUserById(anyString());
+
+        // when
+        var result = mockMvc.perform(get("/users/u/{id}", "non-existent-id"));
+
+        // then
+        result.andExpect(status().isNotFound());
+    }
+
+    @Test
+    @WithMockUser("anonymousUser")
+    void whenUserExistsReturnUserLoggedOut() throws Exception {
```

Comment:
The functions name could be better as it made me thought the user will get logged out or a logged out user is returned.